WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -161,6 +161,19 @@
       patch :update, id: registration.id, registration: { status: :accepted }
       expect(registration.reload.accepted?).to eq true
     end
+
+    it "can register for their own competition that is not yet visible" do
+      competition.update_column(:showAtAll, false)
+      competition.save!
+      expect(RegistrationsMailer).to receive(:notify_organizers_of_new_registration).and_call_original
+      expect(RegistrationsMailer).to receive(:notify_registrant_of_new_registration).and_call_original
+      expect do
+        post :create, competition_id: competition.id, registration: { registration_events_attributes: [ {event_id: "333"} ], guests: 1, comments: "" }
+      end.to change { ActionMailer::Base.deliveries.length }.by(2)
+
+      registration = Registration.find_by_user_id(organizer.id)
+      expect(registration.competitionId).to eq competition.id
```

Comment:
Just a loose suggestion: maybe it would be more expressive to check whether the competition registrations include the registration.